backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/TaskServiceInMemory.java

```diff
@@ -29,8 +25,22 @@ public List<ScriptTask> getScripts() {
 
   @Override
   public String submit(Task task) {
-    tasks.put(task.getId(), task);
-    executorService.submit(task);
+    if (task.getParentTask() != null && tasks.containsKey(task.getParentTask().getId())) {
+      Task parentTask = tasks.get(task.getParentTask().getId());
+      parentTask.addSubTask(task);
+      if (parentTask.getStatus() != RUNNING) {
+        String errorMessage =
+            "Child task started but parent task was not running with status: "
+                + parentTask.getStatus();
+        task.setError(errorMessage);
```

Comment:
so this subtask will be added to parent task. Will parent task complete when all subtasks are completed? or will there need to be a script in the main task to coordinate?